File: client/iih_stub/iih.ci_stub/iih.ci.mrqc/mrtask/d/MrTaskDO.cs

using System;
using System.Text;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.mrqc.mrtask.d
{
    /// <summary>
    /// MrTaskDO 的摘要说明。
    /// </summary>
    public class MrTaskDO : BaseDO
    {

        public MrTaskDO()
        {
        }
        public string Id_mrtask
        {
            get { return getAttrVal<string>("Id_mrtask", null); }
            set { setAttrVal<string>("Id_mrtask", value); }
        }
        public string Id_grp
        {
            get { return getAttrVal<string>("Id_grp", null); }
            set { setAttrVal<string>("Id_grp", value); }
        }
        public string Id_org
        {
            get { return getAttrVal<string>("Id_org", null); }
            set { setAttrVal<string>("Id_org", value); }
        }
        public string Mrtask_name
        {
            get { return getAttrVal<string>("Mrtask_name", null); }
            set { setAttrVal<string>("Mrtask_name", value); }
        }
        public string Id_qa_itm_config
        {
            get { return getAttrVal<string>("Id_qa_itm_config", null); }
            set { setAttrVal<string>("Id_qa_itm_config", value); }
        }
        public string Id_mrtask_sta
        {
            get { return getAttrVal<string>("Id_mrtask_sta", null); }
            set { setAttrVal<string>("Id_mrtask_sta", value); }
        }
        public string Sd_mrtask_sta
        {
            get { return getAttrVal<string>("Sd_mrtask_sta", null); }
            set { setAttrVal<string>("Sd_mrtask_sta", value); }
        }
        public string Id_pat
        {
            get { return getAttrVal<string>("Id_pat", null); }
            set { setAttrVal<string>("Id_pat", value); }
        }
        public string Id_ent
        {
            get { return getAttrVal<string>("Id_ent", null); }
            set { setAttrVal<string>("Id_ent", value); }
        }
        public string Id_exec_dept
        {
            get { return getAttrVal<string>("Id_exec_dept", null); }
            set { setAttrVal<string>("Id_exec_dept", value); }
        }
        public string Id_exec_doctor
        {
            get { return getAttrVal<string>("Id_exec_doctor", null); }
            set { setAttrVal<string>("Id_exec_doctor", value); }
        }
        public string Id_mrcactm
        {
            get { return getAttrVal<string>("Id_mrcactm", null); }
            set { setAttrVal<string>("Id_mrcactm", value); }
        }
        public string Id_mrtp
        {
            get { return getAttrVal<string>("Id_mrtp", null); }
            set { setAttrVal<string>("Id_mrtp", value); }
        }
        public string Id_source
        {
            get { return getAttrVal<string>("Id_source", null); }
            set { setAttrVal<string>("Id_source", value); }
        }
        public string Source_type
        {
            get { return getAttrVal<string>("Source_type", null); }
            set { setAttrVal<string>("Source_type", value); }
        }
        public string Source_name
        {
            get { return getAttrVal<string>("Source_name", null); }
            set { setAttrVal<string>("Source_name", value); }
        }
        public DateTime? Dt_begin
        {
            get { return getAttrVal<FDateTime>("Dt_begin", null); }
            set { setAttrVal<FDateTime>("Dt_begin", value); }
        }
        public DateTime? Dt_end
        {
            get { return getAttrVal<FDateTime>("Dt_end", null); }
            set { setAttrVal<FDateTime>("Dt_end", value); }
        }
        public DateTime? Dt_complete
        {
            get { return getAttrVal<FDateTime>("Dt_complete", null); }
            set { setAttrVal<FDateTime>("Dt_complete", value); }
        }
        public string Time_has
        {
            get
            {
                if (Dt_end != null || Dt_begin != null)
                {
                    TimeSpan timepan = Dt_end.Value - DateTime.Now;

                    if (timepan.TotalDays > 0)
                    {
                        String s = timepan.Hours == 0
                            ? string.Format("{0}分钟", timepan.Minutes)
                            : string.Format("{0}小时{1}分钟", timepan.Hours, timepan.Minutes);
                        StringBuilder sb = new StringBuilder(s);
                        if (timepan.Days > 0)
                        {
                            sb.Insert(0, string.Format("{0}天", timepan.Days));
                        }
                        return getAttrVal<string>("Time_has", sb.ToString());

                    }
                    if (timepan.TotalDays < 0)
                        return getAttrVal<string>("Time_has", "已超时！");
                }

                return getAttrVal<string>("Time_has", null);
            }
            set { setAttrVal<string>("Time_has", value); }
        }
        public string Reason_cancel
        {
            get { return getAttrVal<string>("Reason_cancel", null); }
            set { setAttrVal<string>("Reason_cancel", value); }
        }
        public string Createdby
        {
            get { return getAttrVal<string>("Createdby", null); }
            set { setAttrVal<string>("Createdby", value); }
        }
        public DateTime? Createdtime
        {
            get { return getAttrVal<FDateTime>("Createdtime", null); }
            set { setAttrVal<FDateTime>("Createdtime", value); }
        }
        public string Modifiedby
        {
            get { return getAttrVal<string>("Modifiedby", null); }
            set { setAttrVal<string>("Modifiedby", value); }
        }
        public DateTime? Modifiedtime
        {
            get { return getAttrVal<FDateTime>("Modifiedtime", null); }
            set { setAttrVal<FDateTime>("Modifiedtime", value); }
        }
        public bool? Israte
        {
            get { return getAttrVal<FBoolean>("Israte", null); }
            set { setAttrVal<FBoolean>("Israte", value); }
        }
        public string Rate
        {
            get { return getAttrVal<string>("Rate", null); }
            set { setAttrVal<string>("Rate", value); }
        }
        public string Id_or
        {
            get { return getAttrVal<string>("Id_or", null); }
            set { setAttrVal<string>("Id_or", value); }
        }
        public string Mrtasksta_name
        {
            get { return getAttrVal<string>("Mrtasksta_name", null); }
            set { setAttrVal<string>("Mrtasksta_name", value); }
        }
        public string Mrtasksta_code
        {
            get { return getAttrVal<string>("Mrtasksta_code", null); }
            set { setAttrVal<string>("Mrtasksta_code", value); }
        }
        public string Pat_name
        {
            get { return getAttrVal<string>("Pat_name", null); }
            set { setAttrVal<string>("Pat_name", value); }
        }
        public string Mrcactm_name
        {
            get { return getAttrVal<string>("Mrcactm_name", null); }
            set { setAttrVal<string>("Mrcactm_name", value); }
        }
        public string Mrcactm_code
        {
            get { return getAttrVal<string>("Mrcactm_code", null); }
            set { setAttrVal<string>("Mrcactm_code", value); }
        }
        public string Mrtp_code
        {
            get { return getAttrVal<string>("Mrtp_code", null); }
            set { setAttrVal<string>("Mrtp_code", value); }
        }
        public string Mrtp_name
        {
            get { return getAttrVal<string>("Mrtp_name", null); }
            set { setAttrVal<string>("Mrtp_name", value); }
        }
        public int Ds
        {
            get { return getAttrVal<int>("Ds", 0); }
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv
        {
            get { return getAttrVal<FDateTime>("Sv", null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }

        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_qa_mrtask";
        }

        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_mrtask";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.mrqc.mrtask.d.MrTaskDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[] { "Ds", "Sv", "Id_mrtask", "Id_grp", "Id_org", "Mrtask_name", "Id_qa_itm_config", "Id_mrtask_sta", "Sd_mrtask_sta", "Id_pat", "Id_ent", "Id_exec_dept", "Id_exec_doctor", "Id_mrcactm", "Id_mrtp", "Id_source", "Source_type", "Source_name", "Dt_begin", "Dt_end", "Dt_complete", "Time_has", "Reason_cancel", "Createdby", "Createdtime", "Modifiedby", "Modifiedtime", "Israte", "Rate", "Id_or", "Mrtasksta_name", "Mrtasksta_code", "Pat_name", "Mrcactm_name", "Mrcactm_code", "Mrtp_code", "Mrtp_name" };
        }

    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mrqc/mrtask/d/MrTaskDO.java
package iih.ci.mrqc.mrtask.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.mrqc.mrtask.d.desc.MrTaskDODesc;
import java.math.BigDecimal;

/**
 * 病历任务 DO数据 
 * 
 */
public class MrTaskDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_MRTASK= "Id_mrtask";
	public static final String ID_GRP= "Id_grp";
	public static final String ID_ORG= "Id_org";
	public static final String MRTASK_NAME= "Mrtask_name";
	public static final String ID_QA_ITM_CONFIG= "Id_qa_itm_config";
	public static final String ID_MRTASK_STA= "Id_mrtask_sta";
	public static final String SD_MRTASK_STA= "Sd_mrtask_sta";
	public static final String ID_PAT= "Id_pat";
	public static final String ID_ENT= "Id_ent";
	public static final String ID_EXEC_DEPT= "Id_exec_dept";
	public static final String ID_EXEC_DOCTOR= "Id_exec_doctor";
	public static final String ID_MRCACTM= "Id_mrcactm";
	public static final String ID_MRTP= "Id_mrtp";
	public static final String ID_SOURCE= "Id_source";
	public static final String SOURCE_TYPE= "Source_type";
	public static final String SOURCE_NAME= "Source_name";
	public static final String DT_BEGIN= "Dt_begin";
	public static final String DT_END= "Dt_end";
	public static final String DT_COMPLETE= "Dt_complete";
	public static final String TIME_HAS= "Time_has";
	public static final String REASON_CANCEL= "Reason_cancel";
	public static final String CREATEDBY= "Createdby";
	public static final String CREATEDTIME= "Createdtime";
	public static final String MODIFIEDBY= "Modifiedby";
	public static final String MODIFIEDTIME= "Modifiedtime";
	public static final String ISRATE= "Israte";
	public static final String RATE= "Rate";
	public static final String ID_OR= "Id_or";
	public static final String MRTASKSTA_NAME= "Mrtasksta_name";
	public static final String MRTASKSTA_CODE= "Mrtasksta_code";
	public static final String PAT_NAME= "Pat_name";
	public static final String MRCACTM_NAME= "Mrcactm_name";
	public static final String MRCACTM_CODE= "Mrcactm_code";
	public static final String MRTP_CODE= "Mrtp_code";
	public static final String MRTP_NAME= "Mrtp_name";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_mrtask() {
		return ((String) getAttrVal("Id_mrtask"));
	}	
	public void setId_mrtask(String Id_mrtask) {
		setAttrVal("Id_mrtask", Id_mrtask);
	}
	public String getId_grp() {
		return ((String) getAttrVal("Id_grp"));
	}	
	public void setId_grp(String Id_grp) {
		setAttrVal("Id_grp", Id_grp);
	}
	public String getId_org() {
		return ((String) getAttrVal("Id_org"));
	}	
	public void setId_org(String Id_org) {
		setAttrVal("Id_org", Id_org);
	}
	public String getMrtask_name() {
		return ((String) getAttrVal("Mrtask_name"));
	}	
	public void setMrtask_name(String Mrtask_name) {
		setAttrVal("Mrtask_name", Mrtask_name);
	}
	public String getId_qa_itm_config() {
		return ((String) getAttrVal("Id_qa_itm_config"));
	}	
	public void setId_qa_itm_config(String Id_qa_itm_config) {
		setAttrVal("Id_qa_itm_config", Id_qa_itm_config);
	}
	public String getId_mrtask_sta() {
		return ((String) getAttrVal("Id_mrtask_sta"));
	}	
	public void setId_mrtask_sta(String Id_mrtask_sta) {
		setAttrVal("Id_mrtask_sta", Id_mrtask_sta);
	}
	public String getSd_mrtask_sta() {
		return ((String) getAttrVal("Sd_mrtask_sta"));
	}	
	public void setSd_mrtask_sta(String Sd_mrtask_sta) {
		setAttrVal("Sd_mrtask_sta", Sd_mrtask_sta);
	}
	public String getId_pat() {
		return ((String) getAttrVal("Id_pat"));
	}	
	public void setId_pat(String Id_pat) {
		setAttrVal("Id_pat", Id_pat);
	}
	public String getId_ent() {
		return ((String) getAttrVal("Id_ent"));
	}	
	public void setId_ent(String Id_ent) {
		setAttrVal("Id_ent", Id_ent);
	}
	public String getId_exec_dept() {
		return ((String) getAttrVal("Id_exec_dept"));
	}	
	public void setId_exec_dept(String Id_exec_dept) {
		setAttrVal("Id_exec_dept", Id_exec_dept);
	}
	public String getId_exec_doctor() {
		return ((String) getAttrVal("Id_exec_doctor"));
	}	
	public void setId_exec_doctor(String Id_exec_doctor) {
		setAttrVal("Id_exec_doctor", Id_exec_doctor);
	}
	public String getId_mrcactm() {
		return ((String) getAttrVal("Id_mrcactm"));
	}	
	public void setId_mrcactm(String Id_mrcactm) {
		setAttrVal("Id_mrcactm", Id_mrcactm);
	}
	public String getId_mrtp() {
		return ((String) getAttrVal("Id_mrtp"));
	}	
	public void setId_mrtp(String Id_mrtp) {
		setAttrVal("Id_mrtp", Id_mrtp);
	}
	public String getId_source() {
		return ((String) getAttrVal("Id_source"));
	}	
	public void setId_source(String Id_source) {
		setAttrVal("Id_source", Id_source);
	}
	public String getSource_type() {
		return ((String) getAttrVal("Source_type"));
	}	
	public void setSource_type(String Source_type) {
		setAttrVal("Source_type", Source_type);
	}
	public String getSource_name() {
		return ((String) getAttrVal("Source_name"));
	}	
	public void setSource_name(String Source_name) {
		setAttrVal("Source_name", Source_name);
	}
	public FDateTime getDt_begin() {
		return ((FDateTime) getAttrVal("Dt_begin"));
	}	
	public void setDt_begin(FDateTime Dt_begin) {
		setAttrVal("Dt_begin", Dt_begin);
	}
	public FDateTime getDt_end() {
		return ((FDateTime) getAttrVal("Dt_end"));
	}	
	public void setDt_end(FDateTime Dt_end) {
		setAttrVal("Dt_end", Dt_end);
	}
	public FDateTime getDt_complete() {
		return ((FDateTime) getAttrVal("Dt_complete"));
	}	
	public void setDt_complete(FDateTime Dt_complete) {
		setAttrVal("Dt_complete", Dt_complete);
	}
	public String getTime_has() {
		return ((String) getAttrVal("Time_has"));
	}	
	public void setTime_has(String Time_has) {
		setAttrVal("Time_has", Time_has);
	}
	public String getReason_cancel() {
		return ((String) getAttrVal("Reason_cancel"));
	}	
	public void setReason_cancel(String Reason_cancel) {
		setAttrVal("Reason_cancel", Reason_cancel);
	}
	public String getCreatedby() {
		return ((String) getAttrVal("Createdby"));
	}	
	public void setCreatedby(String Createdby) {
		setAttrVal("Createdby", Createdby);
	}
	public FDateTime getCreatedtime() {
		return ((FDateTime) getAttrVal("Createdtime"));
	}	
	public void setCreatedtime(FDateTime Createdtime) {
		setAttrVal("Createdtime", Createdtime);
	}
	public String getModifiedby() {
		return ((String) getAttrVal("Modifiedby"));
	}	
	public void setModifiedby(String Modifiedby) {
		setAttrVal("Modifiedby", Modifiedby);
	}
	public FDateTime getModifiedtime() {
		return ((FDateTime) getAttrVal("Modifiedtime"));
	}	
	public void setModifiedtime(FDateTime Modifiedtime) {
		setAttrVal("Modifiedtime", Modifiedtime);
	}
	public FBoolean getIsrate() {
		return ((FBoolean) getAttrVal("Israte"));
	}	
	public void setIsrate(FBoolean Israte) {
		setAttrVal("Israte", Israte);
	}
	public String getRate() {
		return ((String) getAttrVal("Rate"));
	}	
	public void setRate(String Rate) {
		setAttrVal("Rate", Rate);
	}
	public String getId_or() {
		return ((String) getAttrVal("Id_or"));
	}	
	public void setId_or(String Id_or) {
		setAttrVal("Id_or", Id_or);
	}
	public String getMrtasksta_name() {
		return ((String) getAttrVal("Mrtasksta_name"));
	}	
	public void setMrtasksta_name(String Mrtasksta_name) {
		setAttrVal("Mrtasksta_name", Mrtasksta_name);
	}
	public String getMrtasksta_code() {
		return ((String) getAttrVal("Mrtasksta_code"));
	}	
	public void setMrtasksta_code(String Mrtasksta_code) {
		setAttrVal("Mrtasksta_code", Mrtasksta_code);
	}
	public String getPat_name() {
		return ((String) getAttrVal("Pat_name"));
	}	
	public void setPat_name(String Pat_name) {
		setAttrVal("Pat_name", Pat_name);
	}
	public String getMrcactm_name() {
		return ((String) getAttrVal("Mrcactm_name"));
	}	
	public void setMrcactm_name(String Mrcactm_name) {
		setAttrVal("Mrcactm_name", Mrcactm_name);
	}
	public String getMrcactm_code() {
		return ((String) getAttrVal("Mrcactm_code"));
	}	
	public void setMrcactm_code(String Mrcactm_code) {
		setAttrVal("Mrcactm_code", Mrcactm_code);
	}
	public String getMrtp_code() {
		return ((String) getAttrVal("Mrtp_code"));
	}	
	public void setMrtp_code(String Mrtp_code) {
		setAttrVal("Mrtp_code", Mrtp_code);
	}
	public String getMrtp_name() {
		return ((String) getAttrVal("Mrtp_name"));
	}	
	public void setMrtp_name(String Mrtp_name) {
		setAttrVal("Mrtp_name", Mrtp_name);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_mrtask";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_qa_mrtask";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(MrTaskDODesc.class);
	}
	
}